pandas/core/internals/concat.py

```diff
@@ -267,6 +273,8 @@ def _concat_managers_axis0(
 
     offset = 0
     blocks = []
+    refs = []
```

Comment:
```suggestion
    refs: list[weakref.ref | None] = []
```

I don't know if that would help avoid having to "cast" it below, but I see this is sometimes done in managers.py (I suppose for that reason)

Reply:
Ah thx, that seemed to work